File: runtime/go/units/wasi/fd_fdstat_set_flags.go
// Only APPEND is acted on (fd_write reads it back and seeks to end); storing it here is what lets a guest turn append mode off at runtime, which an O_APPEND
// OS handle could not.
// Any open fd is accepted, stdio included: toywasm's WASI setup sets NONBLOCK on stdin and treats failure as fatal, so wasmtime's regular-files-only EBADF answer is not copied.
func (w *WASI) wasi_fd_fdstat_set_flags(fd, flags uint32) uint32 {
    m, ok := w.meta[fd]
    if !ok {
        return wasiBadf
    }
    m.fdflags = uint16(flags)
    return wasiOk
}

File: crates/dewasm-backend-ruby/units/wasi/fd_fdstat_set_flags.rb
# Any open fd is accepted, stdio included: toywasm's WASI setup sets NONBLOCK on stdin and treats failure as fatal, so wasmtime's regular-files-only EBADF answer is not copied.
def wasi_fd_fdstat_set_flags(fd, flags)
  meta = @fd_meta[fd]
  io = @fds[fd]
  return ERRNO_BADF if io.nil? || io.is_a?(WasiDir) || meta.nil?
  # Store the fdflags word; APPEND is honored by fd_write (which seeks to end when the bit is set), so clearing it here disables append mode.
  meta[2] = flags & 0x1f
  ERRNO_SUCCESS
end
